src/geometry/Lie_group/lie_group.lean

```diff
@@ -0,0 +1,224 @@
+/-
+{E : Type*} [normed_group E] [normed_space 𝕜 E]  {I : model_with_corners 𝕜 E E}
+{G : Type*} [topological_space G] [charted_space E G] [smooth_manifold_with_corners I G] [group G]
+[h : Lie_group I G]
+{E' : Type*} [normed_group E'] [normed_space 𝕜 E'] [finite_dimensional 𝕜 E']
+{I' : model_with_corners 𝕜 E' E'}
+{G' : Type*} [topological_space G'] [charted_space E' G'] [smooth_manifold_with_corners I' G']
+[group G'] [h' : Lie_group I' G'] : Lie_group (I.prod I') (G×G') :=
+{ smooth_mul := ((smooth_fst.comp smooth_fst).mul (smooth_fst.comp smooth_snd)).prod_mk
+    ((smooth_snd.comp smooth_fst).mul (smooth_snd.comp smooth_snd)),
+  smooth_inv := smooth_fst.inv.prod_mk smooth_snd.inv, } -/
+
+end
+
+section
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+
+/-- Morphism of additive Lie groups. -/
+structure Lie_add_group_morphism (I : model_with_corners 𝕜 E E) (I' : model_with_corners 𝕜 E' E')
+(G : Type*) [topological_space G] [charted_space E G] [smooth_manifold_with_corners I G]
+[add_group G] [topological_add_group G] [Lie_add_group I G]
+(G' : Type*) [topological_space G'] [charted_space E' G'] [smooth_manifold_with_corners I' G']
+[add_group G'] [topological_add_group G'] [Lie_add_group I' G'] extends add_monoid_hom G G' :=
+  (smooth_to_fun : smooth I I' to_fun)
+
+/-- Morphism of Lie groups. -/
+@[to_additive Lie_add_group_morphism]
+structure Lie_group_morphism (I : model_with_corners 𝕜 E E) (I' : model_with_corners 𝕜 E' E')
+(G : Type*) [topological_space G] [charted_space E G] [smooth_manifold_with_corners I G] [group G]
+[topological_group G] [Lie_group I G]
+(G' : Type*) [topological_space G'] [charted_space E' G'] [smooth_manifold_with_corners I' G']
+[group G'] [topological_group G'] [Lie_group I' G'] extends monoid_hom G G' :=
+  (smooth_to_fun : smooth I I' to_fun)
+
+variables {I : model_with_corners 𝕜 E E} {I' : model_with_corners 𝕜 E' E'}
+{G : Type*} [topological_space G] [charted_space E G] [smooth_manifold_with_corners I G]
+[group G] [topological_group G] [Lie_group I G]
+{G' : Type*} [topological_space G'] [charted_space E' G'] [smooth_manifold_with_corners I' G']
+[group G'] [topological_group G'] [Lie_group I' G']
+
+@[to_additive]
+instance : has_one (Lie_group_morphism I I' G G') := ⟨⟨1, smooth_const⟩⟩
+
+@[to_additive]
+instance : inhabited (Lie_group_morphism I I' G G') := ⟨1⟩
+
+@[to_additive]
+instance : has_coe_to_fun (Lie_group_morphism I I' G G') := ⟨_, λ a, a.to_fun⟩
+
+end
+
+end Lie_group
+
+section Lie_group_core
+
+/-- Sometimes one might want to define a Lie additive group `G` without having proved previously
+that `G` is a topological additive group. In such case it is possible to use `Lie_add_group_core`
+that does not require such instance, and then get a Lie group by invoking `to_Lie_ad_group`. -/
+@[nolint has_inhabited_instance]
```

Comment:
I know this linter can feel annoying, but actually it can have a great pedagogical value. It can also help a lot to find out API holes: if you can't easily prove that real numbers form an additive Lie group for instance then you cannot really say you've formalized Lie groups.

Reply:
Done ;)